### DISCUSSION TITLE
How to disable gzip compaction?

### DISCUSSION THREAD

**Discussion by gqqnbig** [seaweedfs/seaweedfs#3108] [General]
I mount seaweedfs to my file system and I've uploaded some files to the mount point. The size reported by ls -l is 164G.

However df reports I only used 6.7G.

```console
$ ll /home/shared-la
total 164G
-rw-rw-r-- 1 gqqnbig gqqnbig  79G 2022-05-27 15:49  a
-rw-rw-r-- 1 gqqnbig gqqnbig  79G 2022-05-27 16:02  b
-rw-rw-r-- 1 gqqnbig gqqnbig 835M 2022-05-27 16:08 'Commitgen Public.zip'
-rw-rw-r-- 1 gqqnbig gqqnbig 5.5G 2022-05-27 16:07  diffs.7z
-rw-rw-r-- 1 gqqnbig gqqnbig 950M 2022-05-27 15:04  diffs-java.7z
$ df -h /home/shared-la
Filesystem           Size  Used Avail Use% Mounted on
172.25.17.14:8888:/  206G  6.7G  199G   4% /home/shared-la
```

Also the web page of the volume server says I only used less than 10GB.

![firefox_ST8Nzp39Ol](https://user-images.githubusercontent.com/614159/170664339-e3748c48-adff-47be-b754-7ccfcfec5db2.png)

What I'm doing is a stress test. I used 

dd if=/dev/zero of=/home/shared-la/a bs=8k count=10000k

to create file `a` and `b`. I guess because the file content is all 0, they are compressed.

I learned Seaweedfs is able to gzip content https://github.com/pipilavvy/seeweedfs-wiki/blob/master/Optimization.md#gzip-content

Can I disable gzip compaction so that I can easily populate junk into the stroage?

**Comment by nivekuil** [comment]
you can pipe /dev/urandom instead of /dev/zero

**Reply by gqqnbig** [reply]
I don't think Seaweedfs knows what parameter I pass into dd. If I use your command, does Seaweedfs still try so hard zipping the file and waste my CPU?

**Reply by nivekuil** [reply]
You can set the MIME type in the write request to application/gzip and seaweedfs should skip any compression checks. I don't think you can do this through mount though, but you should be using the API directly anyway if you need performance. With random data seaweedfs will still try compressing it and then probably see that it isn't worth it, and store the data uncompressed.

**Comment by olavgg** [comment]
I noticed this recently too, as I have ZFS compression I would like to disable compression in seaweedfs, how can I do this?

**Reply by kmlebedev** [reply]
Use cat set empty mime type config file
